sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/kafka/AbstractKafkaPropertiesBeanPostProcessor.java

```diff
@@ -0,0 +1,185 @@
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+package com.azure.spring.cloud.autoconfigure.kafka;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import com.azure.spring.cloud.autoconfigure.context.AzureGlobalProperties;
+import com.azure.spring.cloud.autoconfigure.implementation.kafka.AzureKafkaConfigurationUtils;
+import com.azure.spring.cloud.autoconfigure.implementation.kafka.jaas.Jaas;
+import com.azure.spring.cloud.autoconfigure.implementation.kafka.jaas.JaasResolver;
+import com.azure.spring.cloud.core.implementation.properties.PropertyMapper;
+import com.azure.spring.cloud.core.properties.AzureProperties;
+import com.azure.spring.cloud.service.implementation.kafka.AzureKafkaPropertiesUtils;
+import org.slf4j.Logger;
+
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.config.BeanPostProcessor;
+
+import static com.azure.spring.cloud.autoconfigure.implementation.kafka.AzureKafkaConfigurationUtils.LOG_OAUTH_AUTOCONFIGURATION_CONFIGURE;
+import static com.azure.spring.cloud.autoconfigure.implementation.kafka.AzureKafkaConfigurationUtils.LOG_OAUTH_DETAILED_PROPERTY_CONFIGURE;
+import static com.azure.spring.cloud.autoconfigure.implementation.kafka.AzureKafkaConfigurationUtils.SASL_LOGIN_CALLBACK_HANDLER_CLASS_OAUTH;
+import static com.azure.spring.cloud.autoconfigure.implementation.kafka.AzureKafkaConfigurationUtils.SASL_MECHANISM_OAUTH;
+import static com.azure.spring.cloud.autoconfigure.implementation.kafka.AzureKafkaConfigurationUtils.SECURITY_PROTOCOL_CONFIG_SASL;
+import static com.azure.spring.cloud.autoconfigure.implementation.kafka.AzureKafkaConfigurationUtils.configureKafkaUserAgent;
+import static com.azure.spring.cloud.autoconfigure.implementation.kafka.AzureKafkaConfigurationUtils.needConfigureSaslOAuth;
+import static org.apache.kafka.clients.CommonClientConfigs.SECURITY_PROTOCOL_CONFIG;
+import static org.apache.kafka.common.config.SaslConfigs.SASL_JAAS_CONFIG;
+import static org.apache.kafka.common.config.SaslConfigs.SASL_LOGIN_CALLBACK_HANDLER_CLASS;
+import static org.apache.kafka.common.config.SaslConfigs.SASL_MECHANISM;
+
+abstract class AbstractKafkaPropertiesBeanPostProcessor<T> implements BeanPostProcessor {
+
+	protected static final PropertyMapper PROPERTY_MAPPER = new PropertyMapper();
```

Comment:
https://github.com/spring-projects/spring-boot/blob/2b3a236f88809af72d13bdf3a07b2197a660e675/spring-boot-project/spring-boot/src/main/java/org/springframework/boot/context/properties/PropertyMapper.java#L63

Reply:
I don't get it, @saragluna  is this the same `PropertyMapper`? I don't find a public non-args constructor from the one you shared?

and seems we should  use `PropertyMapper.get()` instead of the new operator ?

Reply:
ah, here is the `com.azure.spring.cloud.core.implementation.properties.PropertyMapper;`